src/v/cloud_storage/remote_partition.cc

```diff
@@ -304,13 +303,13 @@ class partition_record_batch_reader_impl final
             unknown_exception_ptr = std::current_exception();
         }
 
-        // The reader may have been left in an indeterminate state.
-        // Re-set the pointer to it to ensure that it will not be reused.
+        // Regardless of which error, the reader may have been left in an
```

Comment:
nit: this confused me for a moment, until I realized that we only fall through here in case of errors: maybe tweak the comment to mention that this path is only taken in case of exceptions from the above try{}

Reply:
Done